crates/sui-types/src/dynamic_field.rs

```diff
@@ -25,10 +26,14 @@ pub struct Field<N, V> {
     pub value: V,
 }
 
+#[serde_as]
 #[derive(Clone, Serialize, Deserialize, JsonSchema, Debug)]
 #[serde(rename_all = "camelCase")]
 pub struct DynamicFieldInfo {
     pub name: DynamicFieldName,
+    #[schemars(with = "Base58")]
+    #[serde_as(as = "Readable<Base58, _>")]
+    pub bcs_name: Vec<u8>,
```

Comment:
n00b q: what is this used for?

Reply:
this is the name of the dynamic field in bcs, `DynamicFieldName` is the parsed json version of this.

This is added so we can use it in `get_table_value` to get the bcs name without reading the Field object.
https://github.com/MystenLabs/sui/blob/417d448aca06890fe509c7bcd4388f2c29b1a1f8/crates/sui-core/src/authority.rs#L1962-L1974

This will also be useful for client with bcs support where they can use the bcs bytes instead of the parsed json dynamic field name.